km/km_filesys.c

```diff
@@ -482,12 +482,23 @@ uint64_t km_fs_fcntl(km_vcpu_t* vcpu, int fd, int cmd, uint64_t arg)
    if (cmd == F_DUPFD || cmd == F_DUPFD_CLOEXEC) {
       if (ret >= 0) {
          km_file_t* file = &km_fs()->guest_files[fd];
-         ret = km_add_guest_fd_internal(vcpu,
-                                        ret,
-                                        km_guestfd_name(vcpu, fd),
-                                        (cmd == F_DUPFD) ? 0 : O_CLOEXEC,
-                                        file->how,
-                                        ops);
+         if (file->sockinfo != NULL) {
+           ret = km_add_socket_fd(vcpu,
```

Comment:
If the duped socket has a local address bound, this function does not add the bind address to the new km_fd_socket_t.